src/main/java/hudson/plugins/promoted_builds/parameters/PromotedBuildParameterDefinition.java

```diff
@@ -113,10 +132,39 @@ public String getProcess() {
         return promotionProcessName;
     }
 
+    /**
+     * Gets a list of promoted builds for the project.
+     * @return List of {@link AbstractBuild}s, which have been promoted
+     * @deprecated This method retrieves the base item for relative addressing from 
+     * the {@link StaplerRequest}. The relative addressing may be malfunctional if
+     * you use this method outside {@link StaplerRequest}s. 
+     * Use {@link #getBuilds(hudson.model.Item)} instead
+     */
+    @Nonnull
+    @Deprecated
     public List getBuilds() {
-        List builds = new ArrayList();
+        // Try to get ancestor from the object, otherwise pass null and disable the relative addressing
+        final StaplerRequest currentRequest = Stapler.getCurrentRequest();
+        final Item item = currentRequest != null ? currentRequest.findAncestorObject(Item.class) : null;
+        return getBuilds(item);
+    }
+    
+    /**
+     * Gets a list of promoted builds for the project.
+     * @param base Base item for the relative addressing
+     * @return List of {@link AbstractBuild}s, which have been promoted
+     * @since TODO
+     */
+    @Nonnull
+    public List getBuilds(@CheckForNull Item base) {
+        final List builds = new ArrayList();
+        final Jenkins jenkins = Jenkins.getInstance();
+        if (jenkins == null) {
+            return builds;
```

Comment:
what builds do you expect return if jenkins is null, i.e. no builds and no code may work?